### PR TITLE
feat(dropdown, dropdown-item, dropdown-group): add component tokens with fallbacks

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #7180 

### Dropdown
--calcite-dropdown-background-color: specifies the background color of the component.

### Dropdown Item
--calcite-dropdown-item-background-color: specifies the background color of the component.
--calcite-dropdown-item-icon-color: specifies the color of an icon in the component.
--calcite-dropdown-item-text-color: specifies the text color of the component.

### Dropdown Group
--calcite-dropdown-group-border-color: specifies the border color of the component.
--calcite-dropdown-group-text-color: specifies the text color of the component.

### Split Button
--calcite-split-button-dropdown-background-color: defines the background color of the component's dropdown.

### CODE REVIEW

File: packages/calcite-components/src/components/dropdown/dropdown.scss [hunk 1]
```diff
@@ -3,7 +3,8 @@
  *
  * These properties can be overridden using the component's tag as selector.
  *
- * @prop --calcite-dropdown-width: Specifies the width of the component's wrapper.
+ * @prop --calcite-dropdown-background-color: specifies the background color of the component.
```
Comment (jcfranco) [inline packages/calcite-components/src/components/dropdown/dropdown.scss:6]: Nitpick: For consistency, `specifies` ➡️ `Specifies`

Applies to other doc.

**Review by jcfranco — COMMENTED** [review]
Looking good! Just had a question regarding stateful CSS props.

File: packages/calcite-components/src/components/action/resources.ts [hunk 2]
```diff
@@ -1,12 +1,16 @@
 export const CSS = {
+  actionIndicator: "action-indicator",
   button: "button",
-  buttonTextVisible: "button--text-visible",
   buttonCompact: "button--compact",
-  indicatorText: "indicator-text",
+  buttonTextVisible: "button--text-visible",
   iconContainer: "icon-container",
+  indicatorText: "indicator-text",
+  indicatorWithIcon: "indicator-with-icon",
```
Comment (jcfranco) [inline packages/calcite-components/src/components/action/resources.ts:8]: Nitpick: `--` should be used as a delimiter to state/modifiers. Also, one could be dropped if either with or without icon is treated as the default (e.g., `indicator` + `indicator--icon`).

File: packages/calcite-components/src/components/dropdown/dropdown.scss [hunk 3]
```diff
@@ -23,16 +24,17 @@
 }
 
 :host .calcite-dropdown-content {
```
Comment (jcfranco) [inline packages/calcite-components/src/components/dropdown/dropdown.scss:26]: Nitpick: `:host` shouldn't be necessary here.

File: packages/calcite-components/src/components/dropdown-item/dropdown-item.scss [hunk 4]
```diff
@@ -1,195 +1,180 @@
...
+  padding-inline-end: theme("margin.auto");
+  padding-inline-start: theme("margin.1");
 }
 
 // item icon
-.dropdown-item-icon {
+// Selection Indicator Icon
+calcite-icon.dropdown-item-icon {
+  --calcite-icon-color: var(
+    --calcite-dropdown-item-indicator-color,
```
Comment (jcfranco) [inline packages/calcite-components/src/components/dropdown-item/dropdown-item.scss:157]: Sidebar: I think I've been implementing this pattern a bit differently (the public prop is resolved first when assigning internal CSS props). Will review and update accordingly.

File: packages/calcite-components/src/components/dropdown-item/dropdown-item.scss [hunk 5]
```diff
@@ -1,195 +1,180 @@
...
 }
-
-:host(:active:not([disabled])) .container {
-  @apply bg-foreground-3;
+.container--l {
+  --calcite-internal-dropdown-item-font-size: var(--calcite-font-size-0);
+  --calcite-internal-dropdown-item-icon-space-x-lg: theme("margin.4");
+  --calcite-internal-dropdown-item-icon-space-x-sm: theme("margin.1");
+  --calcite-internal-dropdown-item-icon-space-x-start: theme("spacing.3");
+  --calcite-internal-dropdown-item-line-height: 1.25rem /* 20px */;
```
Comment (jcfranco) [inline packages/calcite-components/src/components/dropdown-item/dropdown-item.scss:134]: Nitpick: The px-equivalent comments aren't necessary.

File: packages/calcite-components/src/components/action/resources.ts (hunk 2 above)
Comment (alisonailea) [reply packages/calcite-components/src/components/action/resources.ts:8]: These are the old classes. Would you like me to add these as a refactor within this PR?

File: packages/calcite-components/src/components/action/resources.ts [hunk 6]
```diff
@@ -1,12 +1,16 @@
 export const CSS = {
+  actionIndicator: "action-indicator",
```
[the remark this answers is not in the document]
Comment (alisonailea) [reply packages/calcite-components/src/components/action/resources.ts:2]: It was an old class that already existed here but it wasn't in use. I added it back to remove the need for an unnecessary mixin in the styles.

File: packages/calcite-components/src/components/action/resources.ts (hunk 2 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/action/resources.ts:8]: Gotcha. We can do this separately then. Might be good to do a quick audit just to see how we're doing there. Very low priority though.

**Review by jcfranco — APPROVED** [review]
Awesome! Good to merge once `action` changes are rolled back. 🚀

**Pull request merged**